tests/Feature/OpenGraphTest.php

```diff
@@ -0,0 +1,128 @@
+<?php
+/**
+ * OpenGraphTest class file
+ *
+ * @package wp-seo
+ */
+
+namespace Alley\WP\WP_SEO\Tests\Features;
+
+use Alley\WP\WP_SEO\Tests\TestCase;
+use Alley\WP\WP_SEO\Features\Open_Graph;
+
+/**
+ * OpenGraph Test for the Open_Graph class.
+ *
+ * @link https://mantle.alley.com/docs/testing
+ */
+class OpenGraphTest extends TestCase {
+	/**
+	 * Test title.
+	 */
+	public function test_get_title() {
+		$post_id = $this->factory->post
+		->with_meta(
+			[
+				'wp_seo_open_graph_title' => 'Open Graph Title',
+			]
+		)
+		->create();
+		$this->assertEquals( 'Open Graph Title', Open_Graph::get_title( $post_id ) );
+	}
+
+	/**
+	 * Test title w/ fallback.
+	 */
+	public function test_get_title_fallback() {
+		$post_id = $this->factory->post
+		->with_meta(
+			[
+				'wp_seo_open_graph_title' => '',
+			]
+		)
+		->create(
+			[
+				'post_title' => 'Post Title',
+			]
+		);
+		$this->assertEquals( 'Post Title', Open_Graph::get_title( $post_id ) );
+	}
+
+	/**
+	 * Test description.
+	 */
+	public function test_get_description() {
+		$post_id = $this->factory->post
+		->with_meta(
+			[
+				'wp_seo_open_graph_description' => 'Open Graph Description',
+			]
+		)
+		->create();
+		$this->assertEquals( 'Open Graph Description', Open_Graph::get_description( $post_id ) );
+	}
+
+	/**
+	 * Test description w/ fallback.
+	 */
+	public function test_get_description_fallback() {
+		$post_id = $this->factory->post
+		->with_meta(
+			[
+				'wp_seo_open_graph_description' => '',
+			]
+		)
+		->create(
+			[
+				'post_excerpt' => 'Post Content',
+			]
+		);
+		$this->assertEquals( 'Post Content', Open_Graph::get_description( $post_id ) );
+	}
+
+	/**
+	 * Test image.
+	 */
+	public function test_get_image() {
+		$post_id = $this->factory->post
+		->with_meta(
+			[
+				'wp_seo_open_graph_image' => 007,
+			]
+		)
+		->create();
+		$this->assertEquals( 007, Open_Graph::get_image( $post_id ) );
```

Comment:
this is failing because get_image returns the url of the image (or false in this case because the image doesn't exist), not the id. There should be a way to load an image into the media library, then get its url. #cc_testing should be able to help.

Reply:
Thanks! It took me a bit to sort out this but I think I came to a pretty good solution here: [d123964](https://github.com/alleyinteractive/wp-seo/pull/132/commits/d123964b373c725f298ddeb7b0e9a2744ac044c5). Post meta source is not identical but I think it works. I also figured out how to run phpunit tests locally in the process (much better than pushing to the remote branch 🤕 )